src/core/Microsoft.PowerToys.Settings.UI.Lib/ViewModels/PowerLauncherViewModel.cs

```diff
@@ -1,290 +1,284 @@
-            }
-
-            set
-            {
-                if (settings.Properties.IgnoreHotkeysInFullscreen != value)
-                {
-                    settings.Properties.IgnoreHotkeysInFullscreen = value;
-                    UpdateSettings();
-                }
-            }
-        }
-
-        public bool ClearInputOnLaunch
-        {
-            get
-            {
-                return settings.Properties.ClearInputOnLaunch;
-            }
-
-            set
-            {
-                if (settings.Properties.ClearInputOnLaunch != value)
-                {
-                    settings.Properties.ClearInputOnLaunch = value;
-                    UpdateSettings();
-                }
-            }
-        }
-
-        public bool DisableDriveDetectionWarning
-        {
-            get
-            {
-                return settings.Properties.DisableDriveDetectionWarning;
-            }
-
-            set
-            {
-                if (settings.Properties.DisableDriveDetectionWarning != value)
-                {
-                    settings.Properties.DisableDriveDetectionWarning = value;
-                    UpdateSettings();
-                }
-            }
-        }
-    }
-}
+// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Text.Json;
+using Microsoft.PowerToys.Settings.UI.Lib.Helpers;
+
+namespace Microsoft.PowerToys.Settings.UI.Lib.ViewModels
+{
+    public class PowerLauncherViewModel : Observable
+    {
+        private PowerLauncherSettings settings;
+        private GeneralSettings generalSettings;
+
+        public delegate void SendCallback(PowerLauncherSettings settings);
+
+        private readonly SendCallback callback;
+
+        private Func<string, int> SendConfigMSG { get; }
+
+        public PowerLauncherViewModel(Func<string, int> ipcMSGCallBackFunc)
+        {
+            try
+            {
+                callback = (PowerLauncherSettings settings) =>
+                {
+                    // Propagate changes to Power Launcher through IPC
+                    SendConfigMSG(
+                        string.Format("{{ \"powertoys\": {{ \"{0}\": {1} }} }}", PowerLauncherSettings.ModuleName, JsonSerializer.Serialize(settings)));
+                };
+                if (SettingsUtils.SettingsExists(PowerLauncherSettings.ModuleName))
+                {
+                    settings = SettingsUtils.GetSettings<PowerLauncherSettings>(PowerLauncherSettings.ModuleName);
+                }
+                else
+                {
+                    settings = new PowerLauncherSettings();
+                    settings.Properties.OpenPowerLauncher.Alt = true;
+                    settings.Properties.OpenPowerLauncher.Code = 32;
```

Comment:
@laviusmotileng-ms Why are you hardcoding this now?

Reply:
Windows.System.VirtualKey.Space is from a [UWP API](https://docs.microsoft.com/en-us/uwp/api/windows.system.virtualkey?view=winrt-19041) which is not available in [.Net Standard system.windows.input namespace](https://docs.microsoft.com/en-us/dotnet/api/system.windows.input?view=netstandard-2.1). I'll try to see if theres an alternative to that.

Reply:
@laviusmotileng-ms:  https://docs.microsoft.com/en-us/dotnet/api/system.windows.input.keyinterop.virtualkeyfromkey?view=netcore-3.1

Reply:
UWP currently only supports .Net Standard 2.0 (which is the library used for viewmodels). The [System.Windows.Input library ](https://docs.microsoft.com/en-us/dotnet/api/system.windows.input.keyinterop.virtualkeyfromkey?view=netcore-3.1) is for .Net Core >= 3.0 and .Net Framework which are not supported in .Net Standard 2.0.

Reply:
updated it here: https://github.com/microsoft/PowerToys/pull/5768/commits/fbf39597021fe4cad3c65bf6589b64a3008cfbb8